tools/run_tests/run_tests.py

```diff
@@ -1067,7 +1067,7 @@ def test_specs(self):
                 environ=_FORCE_ENVIRON_FOR_WRAPPERS))
         # TODO: replace with run_one_test_bazel.sh when Bazel-Xcode is stable
         out.append(
-            self.config.job_spec(['src/objective-c/tests/run_one_test.sh'],
+            self.config.job_spec(['src/objective-c/tests/run_one_test_bazel.sh'],
```

Comment:
please see 
https://github.com/grpc/grpc/pull/29665/files#diff-26e0bc5bba55be18a1e28b707c1768be5c3de415944115ec3b4eb006b1d39693R1081

While I think they way of passing HOST_PORT_LOCALSSL and HOST_PORT_LOCAL env variables is useful (since for bazel we could only use static ports until now), I think we should not run bazel test via the run_tests.py script and via the run_one_test_with_bazel.sh

Instead we should use the grpc_objc_bazel_test I just introduced. That test already runs the unitTests target with bazel and does that together with other settings we want (reports in UI, results uploaded in bigquery) and in a way that more aligned with our other bazel test jobs. also it makes it easier to use remote caching in the future.

I think we should focus on improving the grpc_objc_bazel_test job (there's plenty of TODOs and possible improvements) and after new tests are added there, we should then remove them from the run_tests.py suite (which is slower, more complicated and has other problems).